pandas/_libs/tslibs/period.pyx

```diff
@@ -154,12 +154,30 @@ cdef inline int get_freq_group(int freq) nogil:
     return (freq // 1000) * 1000
 
 
-@cython.cdivision
+@cython.cdivision(False)  # specifically _dont_ use cdvision GH#19643
```

Comment:
add a comment on why

Reply:
rather does this actually matter now that you are *always* using floor division?

Reply:
> rather does this actually matter now that you are _always_ using floor division?

Yes.  With cdivision `-1 // 7` returns 0.  I find it really counter-intuitive, not at all surprising that this bug slipped through.